crates/gpui/src/platform/windows/window.rs

```diff
@@ -125,17 +127,98 @@ impl WindowsWindowInner {
             hide_title_bar,
             display,
             last_ime_input,
+            fullscreen,
         }
     }
 
     fn is_maximized(&self) -> bool {
-        unsafe { IsZoomed(self.hwnd) }.as_bool()
+        !self.is_fullscreen() && unsafe { IsZoomed(self.hwnd) }.as_bool()
     }
 
     fn is_minimized(&self) -> bool {
         unsafe { IsIconic(self.hwnd) }.as_bool()
     }
 
+    fn is_fullscreen(&self) -> bool {
+        self.fullscreen.borrow().is_fullscreen()
+    }
+
+    async fn toggle_fullscreen(self: Rc<Self>) {
+        let mut state = self.fullscreen.borrow_mut();
+        let (next_state, style, x, y, cx, cy, swp_flags) = match &*state {
```

Comment:
What's going on with this code? Does windows not restore the window size when disabling fullscreen?

Reply:
Windows does not have an explicit fullscreen state; a frameless window that matches the display size is treated as fullscreen. Fullscreening is done by moving the window, so the size must be saved to restore it.

Reply:
Ahhh, I see.